wotcpp/webOfTrust.cpp

```diff
@@ -342,36 +354,49 @@ namespace libwot {
       wotDistance[i] = k_max + 1;
     }
     wotDistance[to] = 0;
-    WotStep* root = new WotStep();
+    
+	// Initial step
+	WotStep* root = new WotStep();
     root->distance = 0;
     root->member = to;
     root->previous = NULL;
     paths.push_back(root);
-    lookup(from, to, 1, k_max, root, &paths, &matchingPaths, wotDistance);
-    // Formatting as vectors
-    for (int i = 0; i < matchingPaths.size(); i++) {
-      std::vector<uint32_t> thePath;
+
+    lookup(from, to, 1, k_max, root, &paths, &matchingPaths, wotDistance) ;
+	
+	result.reserve(matchingPaths.size()) ;
+
+	// Formatting as vectors
+	size_t j = 0 ;
+    for (size_t i = 0; i < matchingPaths.size(); ++i) 
+	{
+      std::vector<uint32_t> thePath(k_max+1) ;
```

Comment:
`std::vector<uint32_t> thePath(k_max+1) ;` will force the size of `thePath`, which should be dynamic. It may have a size between `1` and `k_max + 1`